connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java

```diff
@@ -103,16 +107,35 @@ private ResponseEntity<?> processWebhook(
     return connectorResponse;
   }
 
-  private WebhookResultContext toConnectorVariablesContext(
+  // This will be used to correlate data returned from connector.
+  // In other words, we pass this data to Zeebe.
+  private WebhookTriggerResultContext toConnectorVariablesContext(
       WebhookProcessingResult processedResult) {
     if (processedResult == null) {
-      return new WebhookResultContext(null);
+      return new WebhookTriggerResultContext(null, null);
     }
-    return new WebhookResultContext(
-        new WebhookResultContext.Request(
-            Optional.ofNullable(processedResult.body()).orElse(emptyMap()),
-            Optional.ofNullable(processedResult.headers()).orElse(emptyMap()),
-            Optional.ofNullable(processedResult.params()).orElse(emptyMap()),
-            Optional.ofNullable(processedResult.connectorData()).orElse(emptyMap())));
+    return new WebhookTriggerResultContext(
+        new MappedHttpRequest(
+            Optional.ofNullable(processedResult.request().body()).orElse(emptyMap()),
+            Optional.ofNullable(processedResult.request().headers()).orElse(emptyMap()),
+            Optional.ofNullable(processedResult.request().params()).orElse(emptyMap())),
+        Optional.ofNullable(processedResult.connectorData()).orElse(emptyMap()));
+  }
+
+  // This data will be used to compose a response.
+  // In other words, depending on the response body expression,
+  // this data may be returned to the webhook caller.
+  private WebhookProcessResultContext toConnectorProcessVariablesContext(
+      WebhookProcessingResult processedResult, InboundConnectorResult<?> result) {
+    if (processedResult == null) {
+      return new WebhookProcessResultContext(null, null, null);
+    }
+    return new WebhookProcessResultContext(
+        new MappedHttpRequest(
+            Optional.ofNullable(processedResult.request().body()).orElse(emptyMap()),
+            Optional.ofNullable(processedResult.request().headers()).orElse(emptyMap()),
+            Optional.ofNullable(processedResult.request().params()).orElse(emptyMap())),
+        Optional.ofNullable(processedResult.connectorData()).orElse(emptyMap()),
+        result);
   }
```

Comment:
1. Perhaps there could be a better name for this method, as it is currently somewhat confusing. (`toConnectorProcessResultContext`?)
2. Do I get it correctly that if connector doesn't define a response body expression (returns null by default), then the webhook endpoint will return an empty body? And if so, what do you think about changing the default response body expression to `=correlation` so that we always return the correlation result by default, as we currently do?

Reply:
1. Good point on naming. I'll rename.
2. Almost correct, except we don't return empty body but blank, like `200 OK` and that's all. Considering, user wants to return empty body, they will have to override it with `if ... then ... else {}`.
2.1. Regarding returning `=correlation` - that was one of the first implementations, but we discussed it at one of architecture meetings that we are introducing `correlation` but still returning blank response by default.

Reply:
Got it, thank you for the clarification!